x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -242,24 +278,27 @@ export const flattenFields = (allFields: Fields): Fields => {
   return flatten(allFields);
 };
 
-/* this should match https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/kibana/fields_transformer.go */
-interface TypeMap {
-  [key: string]: string;
-}
-const typeMap: TypeMap = {
-  binary: 'binary',
-  half_float: 'number',
-  scaled_float: 'number',
-  float: 'number',
-  integer: 'number',
-  long: 'number',
-  short: 'number',
-  byte: 'number',
-  text: 'string',
-  keyword: 'string',
-  '': 'string',
-  geo_point: 'geo_point',
-  date: 'date',
-  ip: 'ip',
-  boolean: 'boolean',
+export const createFieldFormatMap = (fields: Fields): FieldFormatMap =>
+  fields.reduce<FieldFormatMap>((acc, field) => {
+    if (field.format || field.pattern) {
```

Comment:
What happens in the `field.pattern` case?

Reply:
https://github.com/elastic/kibana/pull/55892/files#diff-d1d96537568bf3a2f026a9bdc83d5ec5R239 . Check out this test. If only pattern exists it gets added with no id.  Here is the test in beats: https://github.com/elastic/beats/blob/master/libbeat/kibana/fields_transformer_test.go#L363

Reply:
Thank you!